lib/facter/lvm_support.rb

```diff
@@ -17,47 +17,48 @@
 Facter.add('lvm_vgs') do
   confine lvm_support: true
 
-  vgs = Facter::Core::Execution.execute('vgs -o name --noheadings 2>/dev/null', timeout: 30)
-
-  if vgs.nil?
-    setcode { 0 }
-  else
-    vg_list = vgs.split
-    setcode { vg_list.length }
+  if Facter.value(:lvm_support)
```

Comment:
this if should not be required, the `lvm_vgs` fact is already confineed on `lvm_support` (see line 18).

Reply:
indeed, if it would be within the setcode block; I'm currently busy with getting rid off these nasty pieces here and to find a solution which works and passes spec tests